content/en/blog/_posts/2020-12-10-Granular-Control-of-Volume-Permission-Changes.md

```diff
@@ -0,0 +1,55 @@
+
+```yaml
+securityContext:
+  runAsUser: 1000
+  runAsGroup: 3000
+  fsGroup: 2000
+  fsGroupChangePolicy: "OnRootMismatch"
+```
+More details about this can be found on - https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#configure-volume-permission-and-ownership-change-policy-for-pods 
+
+### Allow CSI Drivers to declare support for fsGroup based permissions
+
+While the previous section implied that we ALWAYS recursively change permissions of a volume if Pod has a fsGroup, this is not strictly true. For certain multi-writer volume types, such as NFS or Gluster, we don’t perform recursive permission changes even if the pod has a fsGroup. Other volume types may not even support chown/chmod which are unix-based permission control primitives. 
+
+So how do we know when to apply recursive permission changes and when we shouldn’t? For in-tree storage drivers, this was relatively simple. But for CSI drivers that could span a multitude of platforms and storage types, this problem can be challenging. 
+
+Previously, whenever a CSI volume was mounted to a Pod, Kubernetes would attempt to automatically determine if the permissions and ownership should be modified. These methods were imprecise and could cause issues, outlined previously, depending on the storage type.
+
+The FSGroupPolicy field introduced in the CSIDriver spec allows drivers to explicitly opt in or out of these recursive modifications. By having the CSI driver specify a policy for the backing volumes we can avoid needless modification attempts, reducing mount time and avoiding errors.
+#### CSIDriver FSGroupPolicy API
+
+Three FSGroupPolicy values are available as of Kubernetes 1.20, with more planned for future releases.
+
+- ReadWriteOnceWithFSType - This is the default policy, applied if no FSGroupPolicy is defined, and preserves the behavior from previous Kubernetes releases. Each volume is examined at mount time to determine if permissions should be recursively applied.
+- File - Always attempt to apply permission modifications, regardless of the filesystem type or PersistentVolumeClaim’s access mode.
+- None - Never apply permission modifications.
+
+#### How do I use it?
+The only configuration needed is defining an FSGroupPolicy inside of the CSIDriver’s spec. Once that element is defined, any subsequently mounted volumes will automatically use the defined policy. There’s no additional deployment required!
+**What’s next?
+Depending on feedback and adoption, the Kubernetes team plans to push these implementations to GA in either 1.21 or 1.22. 
+### How can I learn more?
+This feature is explained in more detail in the kubernetes-csi docs here and https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#configure-volume-permission-and-ownership-change-policy-for-pods 
```

Comment:
I am not sure if this is on purpose, but you mention `kubernetes-csi docs here` but here is not attached to any links?